wire-ios-sync-engine/Tests/Source/UserSession/ZMUserSessionTests+EncryptionAtRest.swift

```diff
@@ -64,19 +64,9 @@ final class ZMUserSessionTests_EncryptionAtRest: ZMUserSessionTestsBase {
         activityManager = MockBackgroundActivityManager()
         factory = BackgroundActivityFactory.shared
         factory.activityManager = activityManager
-
-        setUpEARServiceWorkaround()
     }
 
-    /// This workaround is needed because all tests here are based on assumptions
-    /// that the `managedObjectContext` is changed.
-    /// To remove the workaround simply the `mockEARService` should be used instead of
-    /// a real instance of `EARService`.
-    private func setUpEARServiceWorkaround() {
```

Comment:
Thanks, I didn't think about overriding 😅

Reply:
But actually the problem itself is not solved, that the tests are testing a real `EARService` through the user session instead of using a mock.
The `EARService` can be tested itself with dedicated unit tests like `EARServiceTests.swift`.
```swift
    /// This workaround is needed because all tests here are based on assumptions
    /// that the `managedObjectContext` is changed.
    /// To remove the workaround simply the `mockEARService` should be used instead of
    /// a real instance of `EARService`.
```

Reply:
Agree, but I fixed a different issue here and that is that the `coreCryptoStack` was being re-used for multiple user sessions which is not allowed anymore since the user session will tear it down when it's closed.